usaspending_api/etl/subaward_etl.py

```diff
@@ -60,10 +64,18 @@ def load_subawards(submission_attributes, db_cursor):
                 d1_empty_count += 1
             continue
 
+        # Get the agency
+        agency = get_valid_awarding_agency(row)
+
+        if not agency:
+            logger.warn("Subaward number {} cannot find matching agency with toptier code {} and subtier code {}".format(row['subcontract_num'], row['awarding_agency_code'], row['awarding_sub_tier_agency_c']))
+            continue
+
         # Find the award to attach this sub-contract to
         # We perform this lookup by finding the Award containing a transaction with
         # a matching parent award id, piid, and submission attributes
-        award = Award.objects.filter(transaction__submission=submission_attributes,
+        award = Award.objects.filter(awarding_agency=agency,
+                                     transaction__submission=submission_attributes,
                                      transaction__contract_data__piid=row['piid'],
```

Comment:
I know we discussed this at length in the last go-round, but could you just search the award itself for `piid` and `parent_award_id`, thus avoiding the need get into `transaction__contract_data`?